subprojects/gst-plugins-bad/sys/msdk/gstmsdkenc.c

```diff
@@ -1713,47 +1696,33 @@ static gboolean
 import_va_surface_to_msdk (GstMsdkEnc * thiz, GstBuffer * buf,
     MsdkSurface * msdk_surface)
 {
-  GstVideoInfo vinfo;
-  GstVideoMeta *vmeta;
-  GstMsdkMemoryID *msdk_mid = NULL;
+  VASurfaceID va_surface;
+  mfxFrameInfo frame_info = { {0,}, 0, };
   mfxFrameSurface1 *mfx_surface = NULL;
-  VASurfaceID surface;
-  gint i;
+  GstMsdkMemoryID *msdk_mid = NULL;
+  mfxMemId *mfx_mid = NULL;
 
-  surface = gst_va_buffer_get_surface (buf);
-  if (surface == VA_INVALID_SURFACE)
+  va_surface = gst_va_buffer_get_surface (buf);
+  if (va_surface == VA_INVALID_SURFACE)
     return FALSE;
 
-  vinfo = thiz->input_state->info;
-  /* Update offset/stride/size if there is VideoMeta attached to
-   * the buffer */
-  vmeta = gst_buffer_get_video_meta (buf);
-  if (vmeta) {
-    if (GST_VIDEO_INFO_FORMAT (&vinfo) != vmeta->format ||
-        GST_VIDEO_INFO_WIDTH (&vinfo) != vmeta->width ||
-        GST_VIDEO_INFO_HEIGHT (&vinfo) != vmeta->height ||
-        GST_VIDEO_INFO_N_PLANES (&vinfo) != vmeta->n_planes) {
-      GST_ERROR_OBJECT (thiz, "VideoMeta attached to buffer is not matching"
-          "the negotiated width/height/format");
-      return FALSE;
-    }
-    for (i = 0; i < GST_VIDEO_INFO_N_PLANES (&vinfo); ++i) {
-      GST_VIDEO_INFO_PLANE_OFFSET (&vinfo, i) = vmeta->offset[i];
-      GST_VIDEO_INFO_PLANE_STRIDE (&vinfo, i) = vmeta->stride[i];
-    }
-    GST_VIDEO_INFO_SIZE (&vinfo) = gst_buffer_get_size (buf);
-  }
+  mfx_surface = (mfxFrameSurface1 *) g_slice_new0 (mfxFrameSurface1);
```

Comment:
type cast is not needed when using g_slice_new0.

Reply:
> type cast is not needed when using g_slice_new0.

Fixed. Thanks.